alacritty/src/window_context.rs

```diff
@@ -0,0 +1,345 @@
+        }
+
+        // Create context for the Alacritty window.
+        Ok(WindowContext {
+            font_size: config.ui_config.font.size(),
+            notifier: Notifier(loop_tx),
+            terminal,
+            display,
+            suppress_chars: Default::default(),
+            message_buffer: Default::default(),
+            received_count: Default::default(),
+            search_state: Default::default(),
+            event_queue: Default::default(),
+            modifiers: Default::default(),
+            mouse: Default::default(),
+            dirty: Default::default(),
+        })
+    }
+
+    /// Update the terminal window to the latest config.
+    pub fn update_config(&mut self, old_config: &Config, config: &Config) {
+        self.display.update_config(config);
+        self.terminal.lock().update_config(config);
+
+        // Reload cursor if its thickness has changed.
+        if (old_config.cursor.thickness() - config.cursor.thickness()).abs() > f32::EPSILON {
+            self.display.pending_update.set_cursor_dirty();
+        }
+
+        if old_config.ui_config.font != config.ui_config.font {
+            // Do not update font size if it has been changed at runtime.
+            if self.font_size == old_config.ui_config.font.size() {
+                self.font_size = config.ui_config.font.size();
+            }
+
+            let font = config.ui_config.font.clone().with_size(self.font_size);
+            self.display.pending_update.set_font(font);
+        }
+
+        // Update display if padding options were changed.
+        let window_config = &old_config.ui_config.window;
+        if window_config.padding(1.) != config.ui_config.window.padding(1.)
+            || window_config.dynamic_padding != config.ui_config.window.dynamic_padding
+        {
+            self.display.pending_update.dirty = true;
+        }
+
+        // Live title reload.
+        if !config.ui_config.window.dynamic_title
+            || old_config.ui_config.window.title != config.ui_config.window.title
+        {
+            self.display.window.set_title(&config.ui_config.window.title);
+        }
+
+        #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
+        self.display.window.set_wayland_theme(&config.ui_config.colors);
+
+        // Set subpixel anti-aliasing.
+        #[cfg(target_os = "macos")]
+        crossfont::set_font_smoothing(config.ui_config.font.use_thin_strokes);
+
+        // Disable shadows for transparent windows on macOS.
+        #[cfg(target_os = "macos")]
+        self.display.window.set_has_shadow(config.ui_config.background_opacity() >= 1.0);
+
+        // Update hint keys.
+        self.display.hint_state.update_alphabet(config.ui_config.hints.alphabet());
+
+        // Update cursor blinking.
+        let event = Event::new(TerminalEvent::CursorBlinkingChange.into(), None);
+        self.event_queue.push(event.into());
```

Comment:
Pushing in the event queue here is a bit annoying, but since we don't have a processor available we can't directly handle the update.